### DISCUSSION TITLE
Starting kratos courier without message_ttl config as of 0.9.0 automatically abandons messages

### DISCUSSION THREAD

**Discussion by jim-obrien-orig** [ory/kratos#2463] [Q&A] [answered]
The config field is this:
```
courier:
   message_ttl: 1h
```
In 0.8.0-alpha-3, I did not have a value set for this, and messages were sent. With 0.9.0-alpha-3, I noticed that messages were not sent, and had a status of `4` which corresponds to `Abandoned`. 
Looking at this code, I can see that all messages will be marked as abandoned if you don't have `message_ttl` set.

https://github.com/ory/kratos/blob/51a13f712d38a942772b3f4c014971ecb4658d7a/courier/courier_dispatcher.go#L54-L55

This actually took a while to debug. I'm thinking that I'll file an issue that this should be documented that this configuration field is required, or a sane default should be provided in the code.

**Comment by aeneasr** [comment, marked as answer]
According to https://github.com/ory/kratos/blob/b682689dd781d52d164b0c6edaf77ab15c9ef154/embedx/config.schema.json#L1477 the default should be 1h

**Reply by jim-obrien-orig** [reply]
Actually I'm thinking it might not be a default issue, but a timezone issue. I have to set the ttl to 24hrs to prevent the message from being abandoned.  I haven't played around enough with the settings but I'm thinking something is using the local timezone and something else is using UTC.

**Reply by Benehiko** [reply]
that's odd, i believe the database is always using UTC

**Reply by Benehiko** [reply]
maybe the courier doesn't trigger because of mailhog chaos monkey "Jim". Are you experiencing these issues only on local or on a production environment as well?

**Reply by jim-obrien-orig** [reply]
Haha - could be. This is local development. We didn't upgrade on the servers yet because the messages weren't going out. Looking at the `courier_messages` entries, the timestamps are all in the local time, not UTC. Could it be that since a local development environment is not using UTC (whereas servers would/should be), that this issue occurs?

**Reply by aeneasr** [reply]
That's not unlikely!
